src/libcore/sync/atomic.rs

```diff
@@ -923,13 +923,14 @@ impl<T> AtomicPtr<T> {
         unsafe { atomic_swap(self.p.get() as *mut usize, ptr as usize, order) as *mut T }
     }
 
-    /// Stores a value into the pointer if the current value is the same as the `current` value.
+    /// Stores a value into the pointer if the current pointer *address* value is the same
+    /// as the `current` pointer *address* value.
     ///
-    /// The return value is always the previous value. If it is equal to `current`, then the value
-    /// was updated.
+    /// The return value is always the previous value contained in the pointer. If it is equal
+    /// to the value pointed by `current`, then the value was updated.
```

Comment:
This doesn't read right to me. 'the previous value contained in the pointer' sounds to me like it's not returning a pointer but a value type. 'If it is equal to the value pointed by `current` sounds similarly misleading. It's not actually true that if the value pointer to be `current` is the same as the value pointed to be the return value then the pointer was updated. What matters is the value of the pointer itself (the address). It's possible for `current` and `new` to point to two different values that are still equal - those values don't have any affect on what the CAS is doing.